### ISSUE TITLE
aws_emr_cluster swallows JSON config errors

### ISSUE THREAD

**Issue by peay** [terraform-providers/terraform-provider-aws#4247]
### Terraform Version

Terraform v0.10.8, AWS provider 1.14

### Affected Resource(s)

- `aws_emr_cluster`

### Expected Behavior

The `configurations` field allows passing in a JSON configuration for the cluster.

When the JSON is invalid (e.g., parse error), an error should occur during `plan` indicating why the JSON configuration cannot be parsed.

### Actual Behavior

The error is silently ignored, and an EMR cluster with an empty configuration is deployed.

During `apply`, debug logs have:

```
2018-04-18T12:33:54.202+0200 [DEBUG] plugin.terraform-provider-aws_v1.14.1_x4: 2018/04/18 12:33:54 [ERROR] parsing JSON invalid character '"' after object key
2018-04-18T12:33:54.202+0200 [DEBUG] plugin.terraform-provider-aws_v1.14.1_x4: 2018/04/18 12:33:54 [ERR] Error reading body JSON: invalid character '"' after object key
2018-04-18T12:33:54.202+0200 [DEBUG] plugin.terraform-provider-aws_v1.14.1_x4: 2018/04/18 12:33:54 [DEBUG] Expanded EMR Configurations []
```

Notice how these are all `DEBUG` messages, manually formatted to have `[ERR]` in their message. This means the messages are only visible at `DEBUG` level.

Then, Terraform proceeds to apply the change and deploy the cluster. As can be seen from the `Expanded EMR configuration` log line, the configuration will be left empty.

In https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_emr_cluster.go#L1326, the error is not propagated, and the error is silently ignored instead of being propagated.

**Comment by bflad** [comment]
Version 1.30.0 of the AWS provider, releasing later today, will have a new `configurations_json` argument that accepts the actual JSON contents and can correctly determine drift from the EMR API. In an upcoming release we'll mark the existing `configurations` argument as deprecated.

To convert to the new attribute, given this Terraform configuration:

```hcl
resource "aws_emr_cluster" "example" {
  # ... other configuration ...
  configurations = "example.json"
}
```

Update it to:

```hcl
resource "aws_emr_cluster" "example" {
  # ... other configuration ...
  configurations_json = "${file("example.json")}"
}
```

**Comment by bflad** [comment]
This has been released in [version 1.30.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1300-august-02-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
